### PR TITLE
Replace v6 to v7 sdk calls

### PR DESCRIPTION
**Pull request by heiderje-vmware**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Tests have been added that prove the fix is effective or that the feature works.
- [ ] New and existing tests pass locally with the changes.
- [ ] Code follows the style guidelines of this project (PEP8, clean code, linter).
- [ ] A self-review of the code has been done.

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: src/cbc_sdk/connection.py [hunk 1]
```diff
@@ -692,6 +692,8 @@ def select(self, cls, unique_id=None, *args, **kwargs):
         if unique_id is not None:
             return select_instance(self, cls, unique_id, *args, **kwargs)
         else:
+            query_obj = self._perform_query(cls, **kwargs)
+
             return self._perform_query(cls, **kwargs)
```
Comment (avanbrunt-cb) [inline src/cbc_sdk/connection.py:697]: This needs reverted

File: src/cbc_sdk/platform/models/base_alert.yaml [hunk 2]
```diff
@@ -10,7 +10,7 @@ properties:
     - MINOR
     - SERIOUS
     - CRITICAL
-  create_time:
+  detection_timestamp:
```
Comment (avanbrunt-cb) [inline src/cbc_sdk/platform/models/base_alert.yaml:13]: Are the new fields coming in a future commit?

File: src/tests/unit/platform/test_platform_models.py [hunk 3]
```diff
@@ -221,22 +221,22 @@ def test_BaseAlert_dismiss(monkeypatch):
...
     alert.dismiss("Fixed", "Yessir")
     assert _was_called
     assert alert.workflow_.changed_by == "Robocop"
-    assert alert.workflow_.state == "DISMISSED"
-    assert alert.workflow_.remediation == "Fixed"
+    assert alert.workflow_.status == "CLOSED"
+    assert alert.workflow_.closure_reason == "Fixed"
     assert alert.workflow_.comment == "Yessir"
-    assert alert.workflow_.last_update_time == "2019-10-31T16:03:13.951Z"
+    assert alert.workflow_.change_timestamp == "2019-10-31T16:03:13.951Z"
```
Comment (avanbrunt-cb) [inline src/tests/unit/platform/test_platform_models.py:239]: Is there backwards compatible support for workflow? These test asserts should not be modified and instead new tests created for v7 fiels or added as additional asserts

File: src/cbc_sdk/platform/alerts.py [hunk 4]
```diff
@@ -1054,7 +1253,7 @@ def _count(self):
 
         url = self._build_url("/_search")
         request = self._build_request(0, -1)
-        resp = self._cb.post_object(url, body=request)
+        resp = self._cb.rpost_object(url, body=request)
```
Comment (avanbrunt-cb) [inline src/cbc_sdk/platform/alerts.py:1256]: Seems like a typo `rpost_object`

File: src/cbc_sdk/platform/alerts.py [hunk 5]
```diff
@@ -1093,6 +1292,12 @@ def _perform_query(self, from_row=1, max_rows=-1):
 
             results = result.get("results", [])
             for item in results:
+                #self.results_compatibility_mapping(item)
```
Comment (avanbrunt-cb) [inline src/cbc_sdk/platform/alerts.py:1295]: Remove dead code

File: src/cbc_sdk/platform/alerts.py [hunk 6]
```diff
@@ -1106,6 +1311,24 @@ def _perform_query(self, from_row=1, max_rows=-1):
                 still_querying = False
                 break
 
+    def results_compatibility_mapping(self, results):
+        v6_query = type(self)
```
Comment (avanbrunt-cb) [inline src/cbc_sdk/platform/alerts.py:1315]: I'm confused how is this defined as v6 query?

File: examples/platform/alerts.py [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+# *******************************************************
+# *
+# * DISCLAIMER. THIS PROGRAM IS PROVIDED TO YOU "AS IS" WITHOUT
+# * WARRANTIES OR CONDITIONS OF ANY KIND, WHETHER ORAL OR WRITTEN,
+# * EXPRESS OR IMPLIED. THE AUTHOR SPECIFICALLY DISCLAIMS ANY IMPLIED
+# * WARRANTIES OR CONDITIONS OF MERCHANTABILITY, SATISFACTORY QUALITY,
+# * NON-INFRINGEMENT AND FITNESS FOR A PARTICULAR PURPOSE.
+
+"""
+Example script illustrating retrieval of container runtime alerts.
```
Comment (avanbrunt-cb) [inline examples/platform/alerts.py:14]: What is this for as it mentions container_runtime but calls CB_ANALYTICS

**Comment by avanbrunt-cb** [comment]
Can you fill out the PR checklist with your intent for the PR and what will be included versus not included? Also feel free to mark PRs which are not fully ready with [WIP] in the title aka [WIP] Replace v6 to v7 sdk calls

File: src/cbc_sdk/platform/models/base_alert.yaml (hunk 2 above)
Comment (heiderje-vmware) [reply src/cbc_sdk/platform/models/base_alert.yaml:13]: yes

File: src/tests/unit/platform/test_platform_models.py (hunk 3 above)
Comment (heiderje-vmware) [reply src/tests/unit/platform/test_platform_models.py:239]: reverted for now will come back when addressing unit tests

File: examples/platform/alerts.py (hunk 7 above)
Comment (heiderje-vmware) [reply examples/platform/alerts.py:14]: This was me creating a separate set of test examples to run off of. May be expanded or removed in a future release. probably shouldn't be in the repo yet. Will keep on local for now

File: src/cbc_sdk/platform/alerts.py (hunk 6 above)
Comment (heiderje-vmware) [reply src/cbc_sdk/platform/alerts.py:1315]: removed. this is when i was before I was on the correct path

**Pull request merged**